### DISCUSSION TITLE
Sometimes python modules are not installed

### DISCUSSION THREAD

**Discussion by hadjian** [saulpw/visidata#2023] [Q&A] [answered]
Hi Community,

I installed v2.11.1 from pip and I am running Linux. It installed into my home folder at `~/.local/lib/python3.8/site-packages/visidata/`. With some tables, vd shows me an error that it cannot find some module, e.g. `xlrd`.

I don't know the python packaging system well enough, but it doesn't seem to install all modules as a dependency. Also it doesn't seem to check the version, as I had an old version of one module installed and it threw an exception because of an API mismatch.

Is this by design, or am I doing something wrong? From `pip show visidata`:

```
Name: visidata
Version: 2.11.1
Summary: terminal interface for exploring and arranging tabular data
Home-page: https://visidata.org
Author: Saul Pwanson
Author-email: visidata@saul.pw
License: GPLv3
Location: /home/hadjian/.local/lib/python3.8/site-packages
Requires: importlib-metadata, python-dateutil
Required-by:
```

**Comment by anjakefala** [comment, marked as answer]
Hi @hadjian!

We [intentionally do not install the format dependencies](https://github.com/saulpw/visidata/issues/1765#issuecomment-1449452055). VisiData supports a lot of formats, most which an individual user might not be interested in, some which have quite complex dependencies.

Some package mantainers have opted to install a [list of dependencies for the most used formats](https://github.com/visidata/homebrew-vd/blob/master/Formula/visidata.rb). If we had a reliable way to poll users, they might actually appreciate the most common dependencies installed via the PyPI installed. At least making an extras available for them.

Here are [all of the supported formats and the dependencies you install for them](https://www.visidata.org/docs/formats/).

Your other point was about pinning minimum versions. You are right that this would be good practice. We welcome anyone to open a PR adding minimum version supports for any dependency. Throughout my work, I will try to add them, too.

**Reply by hadjian** [reply]
Thanks for the explanation!  If there is a dependency mismatch, I know where to look now.
